src/createNavigationContainer.js

```diff
@@ -41,15 +48,33 @@ export default function createNavigationContainer(Component) {
       }
 
       this.state = {
-        nav: this._isStateful()
-          ? Component.router.getStateForAction(this._initialAction)
-          : null,
+        nav: null,
       };
     }
 
+    _renderLoading() {
+      // don't use a default prop for renderLoading, because it fires an invariant for controlled navigators
+      return this.props.renderLoading ? this.props.renderLoading() : null;
+    }
+    _getStorageKey() {
+      // a null storageKey prop tells the container to NOT persist, so we explicitly check for undefined
+      if (this.props.storageKey === undefined) {
```

Comment:
I think state persistence should be opt-in rather than opt-out -- more apps will *not* use it than those that will.

Reply:
sure. I'll rename the prop to `persistenceStorageKey` and make it default to null